src/queue.rs

```diff
@@ -151,6 +184,52 @@ impl<M: GuestAddressSpace> DescriptorChain<M> {
         Self::read_new(mem, dtable_addr, queue_size, queue_size, index)
     }
 
+    /// Create a `DescriptorChain` from the indirect target descriptor table.
+    pub fn new_from_indirect(&self) -> Result<DescriptorChain<M>, Error> {
```

Comment:
The commit message says `Queue::new_from_indirect`, so that should be fixed.

Would it be possible to add an `Option<DescriptorChain<M>>` field for the indirect descriptor, and add something like

```rust
    if let Some(ref mut indirect) = self.curr_indirect {
        return Some(indirect.next())
    } 
    if self.desc.is_indirect() {
        let indirect_chain = Self::new_from_indirect().ok()?;
        let first = indirect_chain.next();
        self.curr_indirect = Some(indirect_chain);
        return Some(first)
    }

    let curr = self.desc;
    // current code
```

? It can be done in a separate pull request too.